internal/blocksync/reactor.go

```diff
@@ -586,7 +586,7 @@ func (bcR *Reactor) processBlock(first, second *types.Block, firstParts *types.P
 
 	// TODO: same thing for app - but we would need a way to
 	// get the hash without persisting the state
-	state, err = bcR.blockExec.ApplyVerifiedBlock(state, firstID, first)
+	state, err = bcR.blockExec.ApplyVerifiedBlock(state, firstID, first, bcR.pool.MaxPeerHeight())
```

Comment:
just want to check, the intention here is to use `bcR.pool.MaxPeerHeight()` because it sets a lock on the pool, but if you're just reading the value, could the `bcR.pool.maxPeerHeight` be used instead to avoid contention ? (but not too sure about the mechanism to update the max peer height, so that might be why)

Reply:
I don't like the lock either, but we need it (or atomic var) to prevent data races.